src/parallel_parameter_search/walk_optimization.py

```diff
@@ -0,0 +1,508 @@
+        orientation_diff = 0.0
+        # wait till time for test is up or stopping condition has been reached
+        while not rospy.is_shutdown():
+            passed_time = self.sim.get_time() - start_time
+            passed_timesteps = passed_time / self.sim.get_timestep()
+            if passed_timesteps == 0:
+                # edge case with division by zero
+                passed_timesteps = 1
+            if start_speed:
+                if passed_time > 2:
+                    # use real speed
+                    self.set_cmd_vel(x * iteration, y * iteration, yaw * iteration)
+                elif passed_time > 1:
+                    self.set_cmd_vel(x * iteration / 2, y * iteration / 2, yaw * iteration / 2)
+            if passed_time > time_limit:
+                # reached time limit, stop robot
+                self.set_cmd_vel(0, 0, 0)
+
+            if passed_time > time_limit + 2:
+                # robot should have stopped now, evaluate the fitness
+                if cost_time:
+                    return False, 0
+                else:
+                    # return self.compute_cost(x * iteration, y * iteration, yaw * iteration)
+                    early_term, cost = self.compute_cost(x * iteration, y * iteration, yaw * iteration)
+                    return early_term, orientation_diff / passed_timesteps
+
+            # test if the robot has fallen down
+            pos, rpy = self.sim.get_robot_pose_rpy()
+            orientation_diff += abs(rpy[0]) + abs(rpy[1] - self.correct_pitch(x, y, yaw))
+            if abs(rpy[0]) > math.radians(45) or abs(rpy[1]) > math.radians(45) or pos[2] < self.trunk_height/2:
+                # add extra information to trial
+                trial.set_user_attr('early_termination_at', (
+                    x * iteration, y * iteration, yaw * iteration))
+                if cost_time:
+                    return True, 1 - min(1, (passed_time / (time_limit + 2)))
+                else:
+                    # return True, cost
+                    return True, 1 - min(1, (passed_time / (time_limit + 2)))
+            try:
+                if self.walk_as_node:
+                    # give time to other algorithms to compute their responses
+                    # use wall time, as ros time is standing still
+                    time.sleep(0.01)
+                else:
+                    current_time = self.sim.get_time()
+                    joint_command = self.walk.step(current_time - self.last_time, self.current_speed,
+                                                   self.sim.get_imu_msg(),
+                                                   self.sim.get_joint_state_msg(),
+                                                   self.sim.get_pressure_left(), self.sim.get_pressure_right())
+                    self.sim.set_joints(joint_command)
+                    self.last_time = current_time
+                self.sim.step_sim()
+            except:
```

Comment:
What are the possible exceptions here? A better handling or error message might be helpful.